tests/test_server.py

```diff
@@ -56,23 +85,115 @@ def test_server_parse_channel_data(server, rtm_start_fixture):
 def test_server_parse_user_data(server, rtm_start_fixture):
     server.parse_user_data(rtm_start_fixture["users"])
     # Find user by Name
-    userbyname = server.users.find('fakeuser')
-    assert type(userbyname) == User
-    assert userbyname == "fakeuser"
-    assert userbyname != "someotheruser"
+    user_by_name = server.users.find('fakeuser')
+    assert type(user_by_name) == User
+    assert user_by_name == "fakeuser"
+    assert user_by_name != "someotheruser"
     # Find user by ID
-    userbyid = server.users.find('U10CX1234')
-    assert type(userbyid) == User
-    assert userbyid == "fakeuser"
-    assert userbyid.email == 'fakeuser@example.com'
+    user_by_id = server.users.find('U10CX1234')
+    assert type(user_by_id) == User
+    assert user_by_id == "fakeuser"
+    assert user_by_id.email == 'fakeuser@example.com'
     # Don't find invalid user
-    userbyid = server.users.find('invaliduser')
-    assert type(userbyid) != User
+    user_by_id = server.users.find('invaliduser')
+    assert user_by_id is None
 
 
 def test_server_cant_connect(server):
-    with pytest.raises(SlackLoginError):
-        reply = server.ping()
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            responses.POST,
+            "https://slack.com/api/rtm.start",
+            status=403,
+            json={"ok": False}
+        )
+
+        with pytest.raises(SlackConnectionError) as e:
+            server.rtm_connect()
+
+
+def test_reconnect_flag(server, rtm_start_fixture):
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            responses.POST,
+            "https://slack.com/api/rtm.start",
+            status=200,
+            json=rtm_start_fixture
+        )
+
+        server.rtm_connect(auto_reconnect=True)
+        assert server.auto_reconnect is True
+
+        for call in rsps.calls:
+            assert call.request.url in [
+                "https://slack.com/api/rtm.start"
+            ]
+
+
+def test_rtm_reconnect(server, rtm_start_fixture):
+    with responses.RequestsMock() as rsps:
+        rsps.add(
+            responses.POST,
+            "https://slack.com/api/rtm.connect",
+            status=200,
+            json=rtm_start_fixture
+        )
+
+        server.rtm_connect(auto_reconnect=True, reconnect=True, use_rtm_start=False)
+
+        for call in rsps.calls:
+            assert call.request.url in [
+                "https://slack.com/api/rtm.connect"
+            ]
+
+
+def test_rtm_reconnect_timeout_recently_connected(server, rtm_start_fixture):
+    # If reconnected recently, server must wait to reconnect and increment the counter
```

Comment:
Feel free to make these docstrings to tie these to the function for some python-y tooling.